test/test_metadata.py

```diff
@@ -0,0 +1,112 @@
+        self.metadata["multi2"] = self.multi2
+        self.multi3 = ["multi3-value1", "multi3-value2"]
+        self.metadata.set("multi3", self.multi3)
+        self.metadata["~hidden"] = "hidden-value"
+
+    def tearDown(self):
+        pass
+
+    def test_metadata_set(self):
+        self.assertEqual(["single1-value"], dict.get(self.metadata,"single1"))
+        self.assertEqual(["single2-value"], dict.get(self.metadata,"single2"))
+        self.assertEqual(self.multi1, dict.get(self.metadata,"multi1"))
+        self.assertEqual(self.multi2, dict.get(self.metadata,"multi2"))
+        self.assertEqual(self.multi3, dict.get(self.metadata,"multi3"))
+        self.assertEqual(["hidden-value"], dict.get(self.metadata,"~hidden"))
+
+    def test_metadata_get(self):
+        self.assertEqual("single1-value", self.metadata["single1"])
+        self.assertEqual("single1-value", self.metadata.get("single1"))
+        self.assertEqual(["single1-value"], self.metadata.getall("single1"))
+
+        self.assertEqual(MULTI_VALUED_JOINER.join(self.multi1), self.metadata["multi1"])
+        self.assertEqual(MULTI_VALUED_JOINER.join(self.multi1), self.metadata.get("multi1"))
+        self.assertEqual(self.multi1, self.metadata.getall("multi1"))
+
+        self.assertEqual("", self.metadata["nonexistent"])
+        self.assertEqual(None, self.metadata.get("nonexistent"))
+        self.assertEqual([], self.metadata.getall("nonexistent"))
+
+        self.assertEqual(dict.items(self.metadata), self.metadata.rawitems())
+        metadata_items = [(x, z) for (x, y) in dict.items(self.metadata) for z in y]
+        self.assertEqual(metadata_items, list(self.metadata.items()))
+
+    def test_metadata_delete(self):
+        self.metadata.delete("single1")
+        self.assertNotIn("single1", self.metadata)
+        self.assertIn("single1", self.metadata.deleted_tags)
+
+        self.metadata["single2"] = ""
+        self.assertNotIn("single2", self.metadata)
+        self.assertIn("single2", self.metadata.deleted_tags)
+
+    def test_metadata_update(self):
+        m = Metadata()
+        m["old"] = "old-value"
+        self.metadata.delete("single1")
+        m.update(self.metadata)
+        self.assertIn("old", m)
+        self.assertNotIn("single1", m)
+        self.assertIn("single1", m.deleted_tags)
+        self.assertEqual("single2-value", m["single2"])
+        self.assertEqual(self.metadata.deleted_tags, m.deleted_tags)
+
+        self.metadata["old"] = "old-value"
+        for (key, value) in dict.items(self.metadata):
+            self.assertIn(key, m)
+            self.assertEqual(value, dict.get(m, key))
+        for (key, value) in dict.items(m):
+            self.assertIn(key, self.metadata)
+            self.assertEqual(value, dict.get(self.metadata, key))
+
+    def test_metadata_clear(self):
+        self.metadata.clear()
+        self.assertEqual(0, len(self.metadata))
+
+    def test_metadata_applyfunc(self):
+        func = lambda x: x[1:]
```

Comment:
I want to use the same lambda function multiple times - and because it is a function to use as test data and not a function which is part of the testing functionality (if I have explained that well enough), this seemed to be the best approach. But I can change it if you want.